webapp/tests/conftest.py

```diff
@@ -36,21 +33,52 @@ def test_request_context(app):
         yield req
 
 
+CURRENT_USER_IDNR = "0123456789"
+
+
+@pytest.fixture(autouse=True)
+def testing_current_user(monkeypatch):
+    monkeypatch.setattr("app.forms.session_data.current_user", MagicMock(idnr_hashed=CURRENT_USER_IDNR))
+
+
+@pytest.fixture(autouse=True)
+def testing_database(monkeypatch):
+    fakeredis_connection = fakeredis.FakeStrictRedis()
+    monkeypatch.setattr("app.data_access.form_data_controller.FormDataController._redis_connection",
+                        fakeredis_connection)
+    yield
+    fakeredis_connection.flushall()
+
+
 @pytest.fixture
 def new_test_request_context(app):
     @contextmanager
     def _new_test_request_context(method='GET', form_data=None, stored_data=None, session_identifier='form_data'):
+        from app.forms.session_data import serialize_session_data
+        from app.data_access.form_data_controller import FormDataController
         with app.test_request_context() as req:
             req.request.method = method
             if stored_data:
-                req.session = SecureCookieSession({session_identifier: create_session_form_data(stored_data)})
-            if form_data:
-                req.request.data = ImmutableMultiDict(form_data)
-                req.request.form = ImmutableMultiDict(form_data)
+                FormDataController().save_to_redis(CURRENT_USER_IDNR + '_' + session_identifier,
+                                                   serialize_session_data(stored_data))
+            else:
+                FormDataController().save_to_redis(CURRENT_USER_IDNR + '_' + session_identifier,
+                                                   serialize_session_data(form_data))
             yield req
+
```

Comment:
As we do not set the data on the request context, I am wondering if we could use a second fixture (s. below). This would allow us to keep the functionality of the test_request_context the same which we need for steps where we do not use the redis session:
```
@pytest.fixture
 def new_test_request_context_with_data_in_session(app, new_test_request_context):
     @contextmanager
     def _new_test_request_context(method='GET', form_data=None, session_data=None, session_identifier='form_data'):
          req =  new_test_request_context(method=method, form_data)
          if session_data:
                  FormDataController().save_to_redis(CURRENT_USER_IDNR + '_' + session_identifier,
                                                     serialize_session_data(stored_data))
          yield req
```
As this should still create a correct post request, we need to set the form data not in this method but set them as part of the request_context. form_data is the data that the post request sends while stored_data is the data that we already have in the session (in the cookie context it is part of the cookie, in the session context it is part of the session)